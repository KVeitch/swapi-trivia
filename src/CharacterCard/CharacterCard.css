.character__card {
  width:45%;
  border:1px solid gray;
  display: inline-block;
  height:300px;
  padding:8px;
  margin:5px;
  background-color: rgba(194, 127, 120, 0.6);
  border-radius: 5px;
}

.characer__name {
  font-size: 22px;
}

.character__data__list,
.character__films {
  list-style: none;
  margin-top:10px;
}

