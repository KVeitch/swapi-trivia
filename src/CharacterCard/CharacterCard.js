import React from 'react';
import './CharacterCard.css'

const CharacterCard = props => {
  const { films, homeworld, name, population, species} = props.character

  const classList = props.favoriteList && props.favoriteList.includes(name) ? 'character__card favorite' : 'character__card'

  const filmList = films.map(film => <li>{film}</li>)
  return(
    <div className={classList}>
      <h2 className='character__name'>{name}</h2>
      <ul className='character__data__list'>
          <li className='character__species'>Species: {species}</li>
          <li className='character__homeworld'>Homeworld: {homeworld}</li>
          <li className='character__population'>Population: {population}</li>
        </ul>
        <ul className='character__films'>Appearing in:{filmList}</ul>
        <div type='button' className='div__favorite' 
        onClick={()=>props.setFavorite({ films, homeworld, name, population, species }, name)}
        >
          Favorite
        </div>
    </div>
  )
}

export default CharacterCard;