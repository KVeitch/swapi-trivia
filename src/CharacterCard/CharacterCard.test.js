import React from 'react';
import { shallow } from 'enzyme';
import CharacterCard from './CharacterCard';

describe('Character Card', () => {

    it('should match the snapshot with all the data passed in correctly', () => {
        const wrapper =
            shallow(<CharacterCard 
                        key={3}
                        character={''}/>)
        expect(wrapper).toMatchSnapshot();
    });
});