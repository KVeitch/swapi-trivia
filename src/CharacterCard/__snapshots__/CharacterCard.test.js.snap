// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Card should match the snapshot with all the data passed in correctly 1`] = `
<div
  className="character__card"
>
  <h2
    className="characer__name"
  >
    Bob
  </h2>
  <ul
    className="character__data__list"
  >
    <li
      className="character__species"
    >
      Species: 
      humanish
    </li>
    <li
      className="character__homeworld"
    >
      Homeworld: 
      earth
    </li>
    <li
      className="character__population"
    >
      Population: 
      8000000000
    </li>
  </ul>
  <ul
    className="character__films"
  >
    Appearing in:
    <li>
      Best of my Home movies
    </li>
    <li>
      Mostly vines
    </li>
  </ul>
  <div
    className="div__favorite"
    onClick={[Function]}
    type="button"
  >
    Favorite
  </div>
</div>
`;
